android/java/org/chromium/chrome/browser/crypto_wallet/activities/AccountDetailActivity.java

```diff
@@ -62,29 +65,27 @@
         extends BraveWalletBaseActivity implements OnWalletListItemClick, ApprovedTxObserver {
     private static final String TAG = "AccountDetail";
 
-    private String mAddress;
-    private String mName;
-    private boolean mIsImported;
-    private int mCoinType;
-    private String mKeyringId;
+    private AccountInfo mAccountInfo;
     private TextView mAccountText;
     private ExecutorService mExecutor;
     private Handler mHandler;
     private WalletCoinAdapter mWalletTxCoinAdapter;
     private WalletModel mWalletModel;
 
+    public static Intent createIntent(Context context, @NonNull AccountInfo accountInfo) {
+        Intent intent = new Intent(context, AccountDetailActivity.class);
+        WalletUtils.addAccountInfoToIntent(intent, accountInfo);
+        return intent;
+    }
+
     @Override
     protected void triggerLayoutInflation() {
         setContentView(R.layout.activity_account_detail);
 
         mExecutor = Executors.newSingleThreadExecutor();
         mHandler = new Handler(Looper.getMainLooper());
         if (getIntent() != null) {
-            mCoinType = getIntent().getIntExtra(Utils.COIN_TYPE, CoinType.ETH);
-            mKeyringId = getIntent().getStringExtra(Utils.KEYRING_ID);
-            mAddress = getIntent().getStringExtra(Utils.ADDRESS);
-            mName = getIntent().getStringExtra(Utils.NAME);
-            mIsImported = getIntent().getBooleanExtra(Utils.ISIMPORTED, false);
+            mAccountInfo = WalletUtils.getAccountInfoFromIntent(getIntent());
         }
```

Comment:
What is the graceful way to handle missing or corrupted AccountInfo in intent's data?

Reply:
I mean whole activity doesn't make sense if there is no valid account info, so this activity should stop and/or fallback to some default activity or something else. 
@Pavneet-Sing

Reply:
I mean whole activity doesn't make sense if there is no valid account info, so this activity should stop and/or fallback to some default activity or something else.
@Pavneet-Sing

Reply:
Code below will just crash with NPE when there is no account info

Reply:
Ideally, all paths to this Activity should have the required data (missing case would be a bug to fix). If we cannot(or don't want to) proceed without data or default value then use `finish()`(as you discovered) to go back.